### PR TITLE
test: add check dependencies

### PR DESCRIPTION
**Pull request by qiuming520**
check whether 'kind' is installed
Signed-off-by: qiuming520 <qiuming_yewu@cmss.chinamobile.com>

### CODE REVIEW

**Comment by Iceber** [comment]
@wzshiming PTAL, Thanks

File: test/helper.sh [hunk 1]
```diff
@@ -1,7 +1,20 @@
...
 
+# check whether command is installed.
+function cmd_must_exist {
+    local CMD=$(command -v ${1})
+    if [[ ! -x ${CMD} ]]; then
+      echo "error: Please install ${1} and verify they are in \$PATH."
+      exit 1
+    fi
+}
+
```
Comment (wzshiming) [inline test/helper.sh:17]:
```suggestion
# check whether command is installed.
function cmd_must_exist {
    if ! command -v ${1} > /dev/null 2>&1; then
      echo "error: Please install ${1} and verify they are in \$PATH."
      exit 1
    fi
}

```

For builtin and alias commands, use -x will judgment fails. 
We could just use `command -v`.

File: test/helper.sh [hunk 2]
```diff
@@ -14,6 +27,8 @@ function load_image() {
     if ! docker image inspect "${image}" >/dev/null 2>&1; then
         docker pull "${image}"
     fi
+
+    cmd_must_exist "kind"
```
Comment (wzshiming) [inline test/helper.sh:31]: Maybe it's better to add all dependency checks at the beginning of the test, rather than at every location.

**Review by wzshiming — CHANGES_REQUESTED** [review]

**Comment by wzshiming** [comment]
``` bash
# Check a command exist
function command_exist() {
    local command="${1}"
    type "${command}" >/dev/null 2>&1
}

# Check dependencies is installed or not and exit if not
function check_dependencies() {
    local dependencies=("${@}")
    local not_installed=()
    for dependency in "${dependencies[@]}"; do
        if ! command_exist "${dependency}"; then
            not_installed+=("${dependency}")
        fi
    done

    if [ "${#not_installed[@]}" -ne 0 ]; then
        echo "Error: Some dependencies are not installed:"
        for dependency in "${not_installed[@]}"; do
            echo "  - ${dependency}"
        done
        exit 1
    fi
}


check_dependencies docker kind kubectl
```

File: test/helper.sh [hunk 3]
```diff
@@ -1,7 +1,20 @@
 #!/usr/bin/env bash
 
+set -o errexit
```
[the remark this answers is not in the document]
Comment (qiuming520) [reply test/helper.sh]: remove  ‘set -o errexit’

File: test/helper.sh [hunk 4]
```diff
@@ -1,7 +1,20 @@
...
+set -o nounset
+set -o pipefail
```
[the remark this answers is not in the document]
Comment (qiuming520) [reply test/helper.sh]: have removed

File: test/helper.sh (hunk 2 above)
Comment (qiuming520) [reply test/helper.sh]: add 
# check dependency
cmd_must_exist "kind"
cmd_must_exist "docker"
cmd_must_exist "kubectl"
cmd_must_exist "helm"

**Comment by wzshiming** [comment]
https://github.com/clusterpedia-io/clusterpedia/blob/9ef1a04b28763b4ff5a02f7cd1967b5688f28c8a/test/test.sh#L7-L15

This L11 is a good place to check dependencies.

File: test/helper.sh [hunk 5]
```diff
@@ -1,7 +1,31 @@
 #!/usr/bin/env bash
-
 ROOT="$(dirname "${BASH_SOURCE[0]}")/.."
 
+# check whether command is installed.
+function cmd_must_exist() {
```
Comment (wzshiming) [inline test/helper.sh:5]:
```suggestion
function cmd_exist() {
```

**Review by wzshiming — COMMENTED** [review]
Others lgtm

**Review by wzshiming — APPROVED** [review]
/assign @Iceber

lgtm for me

**Review by Iceber — APPROVED** [review]
LGTM, Thanks to @qiuming520  for contributing and @wzshiming  for reviewing

**Comment by Iceber** [comment]
@qiuming520 The commit message is too long, suggest to change it to `test: check dependencies` or other more concise message

**Comment by qiuming520** [comment]
> Here https://github.com/clusterpedia-io/clusterpedia/pull/239/commits also needs to be updated, the commit message is already long enough to be truncated

yeah ！ I have changed~ @Iceber

**Pull request merged**